tests/attr/test_targets.py

```diff
@@ -1,668 +1,205 @@
 #!/usr/bin/env python3
 
-import unittest
+
+from typing import Any, Callable, Dict, List, Tuple, Type, cast
 
 import torch
+from torch import Tensor
+from torch.nn import Module
 
-from captum.attr._core.deep_lift import DeepLift, DeepLiftShap
-from captum.attr._core.feature_ablation import FeatureAblation
-from captum.attr._core.gradient_shap import GradientShap
-from captum.attr._core.input_x_gradient import InputXGradient
+from captum._utils.common import _format_additional_forward_args
+from captum.attr._core.feature_permutation import FeaturePermutation
 from captum.attr._core.integrated_gradients import IntegratedGradients
-from captum.attr._core.layer.internal_influence import InternalInfluence
-from captum.attr._core.layer.layer_conductance import LayerConductance
-from captum.attr._core.layer.layer_deep_lift import LayerDeepLift, LayerDeepLiftShap
-from captum.attr._core.layer.layer_feature_ablation import LayerFeatureAblation
-from captum.attr._core.layer.layer_gradient_x_activation import LayerGradientXActivation
-from captum.attr._core.layer.layer_integrated_gradients import LayerIntegratedGradients
-from captum.attr._core.neuron.neuron_conductance import NeuronConductance
 from captum.attr._core.noise_tunnel import NoiseTunnel
-from captum.attr._core.occlusion import Occlusion
-from captum.attr._core.saliency import Saliency
-from captum.attr._core.shapley_value import ShapleyValueSampling
-
-from ..helpers.basic import BaseGPUTest, BaseTest, assertTensorAlmostEqual
+from captum.attr._utils.attribution import Attribution, InternalAttribution
+
+from ..helpers.basic import (
+    BaseTest,
+    assertTensorTuplesAlmostEqual,
+    deep_copy_args,
+    get_nested_attr,
+)
 from .helpers.basic_models import BasicModel_MultiLayer
+from .helpers.test_config import config
+
+
+class TargetsMeta(type):
+    """
+    Target tests created in TargetsMeta apply to any test case with targets being a
+    list or tensor.
+    Attribution of each example is computed independently with the appropriate target
+    and compared to the corresponding result of attributing to a batch with a tensor
+    / list of targets.
+    """
+
+    def __new__(cls, name: str, bases: Tuple, attrs: Dict):
+        for test_config in config:
+            algorithms = cast(List[Type[Attribution]], test_config["algorithms"])
+            model = test_config["model"]
```

Comment:
It looks like this piece is being repeated in couple places. I wonder if we could generalize it ?